### PR TITLE
CLOUDSTACK-9135 - As a Developer I want the test_internal_lb.py to test Redundant VPCs

### PR DESCRIPTION
**Pull request by wilderrodrigues**
This PR refactors the existing tests and adds tests to cover internal LB with Redundant VPCs.

In order to make it possible I had to change the base.py, allowing multiple providers per service. This change requires the changes in the Java side which were dealt with PR #1215 .

### CODE REVIEW

**Comment by wilderrodrigues** [comment]
Ping @remibergsma @DaanHoogland @borisroman @michaelandersen @bhaisaab @karuturi @miguelaferreira 

Please, keep in mind that this PR depends on PR #1215.

* Environment
  - 1 KVM host on CentOS 7.1
  - Management Server on CentOS 7.1
  - Agent + Common RPMs built from source

* Test executed

```
nosetests --with-marvin --marvin-config=/data/shared/marvin/mct-zone2-kvm2-ISOLATED.cfg -s -a tags=advanced,required_hardware=true smoke/test_internal_lb.py
```

* Results

```
[root@cs2 integration]# less /tmp//MarvinLogs/test_internal_lb_DXYD14/results.txt 
Test create, assign, remove of an Internal LB with roundrobin http traffic to 3 vm's in a Single VPC ... === TestName: test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Status : SUCCESS ===
ok
Test create, assign, remove of an Internal LB with roundrobin http traffic to 3 vm's in a Redundant VPC ... === TestName: test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | Status : SUCCESS ===
ok
Test to verify access to loadbalancer haproxy admin stats page ... === TestName: test_03_vpc_internallb_haproxy_stats_on_all_interfaces | Status : SUCCESS ===
ok
Test to verify access to loadbalancer haproxy admin stats page ... === TestName: test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 4 tests in 6681.151s

OK
/tmp//MarvinLogs/test_internal_lb_DXYD14/results.txt (END)
```

**Comment by borisroman** [comment]
Hi @wilderrodrigues ,

Looks good! Thanks for making the tests compliant!

```
Test create, assign, remove of an Internal LB with roundrobin http traffic to 3 vm's in a Single VPC ... === TestName: test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Status : SUCCESS ===
ok
Test create, assign, remove of an Internal LB with roundrobin http traffic to 3 vm's in a Redundant VPC ... === TestName: test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | Status : SUCCESS ===
ok
Test to verify access to loadbalancer haproxy admin stats page ... === TestName: test_03_vpc_internallb_haproxy_stats_on_all_interfaces | Status : SUCCESS ===
ok
Test to verify access to loadbalancer haproxy admin stats page ... === TestName: test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 4 tests in 5901.365s
```

LGTM :+1:

File: test/integration/smoke/test_internal_lb.py [hunk 1]
```diff
@@ -286,9 +302,12 @@ def setUpClass(cls):
                    %s" % (cls.account.name,
                           cls.account.id))
 
-        cls.cleanup = [cls.template, cls.account, cls.compute_offering]
+        cls._cleanup = [cls.template, cls.account, cls.compute_offering]
```
Comment (borisroman) [inline test/integration/smoke/test_internal_lb.py]: Why use the prefix _ here?

Comment (wilderrodrigues) [inline test/integration/smoke/test_internal_lb.py]: Hi @borisroman

I added a setUp() and tearDown() methods that are called before/after each test. By doing so, we make sure that each test is done in isolation.

The tearDown() method uses the cleanup list, whilst the tearDownClass() method uses the _cleanup list.

The tearDownClass() is called only when all the tests are done.

:)

**Comment by wilderrodrigues** [comment]
Thanks for the reviw, @borisroman! :)

**Comment by remibergsma** [comment]
LGTM based on running the tests:

```
[root@cs1 integration]# cat /tmp//MarvinLogs/test_internal_lb_SFTBEY/results.txt 
Test create, assign, remove of an Internal LB with roundrobin http traffic to 3 vm's in a Single VPC ... === TestName: test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Status : SUCCESS ===
ok
Test create, assign, remove of an Internal LB with roundrobin http traffic to 3 vm's in a Redundant VPC ... === TestName: test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | Status : SUCCESS ===
ok
Test to verify access to loadbalancer haproxy admin stats page ... === TestName: test_03_vpc_internallb_haproxy_stats_on_all_interfaces | Status : SUCCESS ===
ok
Test to verify access to loadbalancer haproxy admin stats page ... === TestName: test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 4 tests in 3847.453s

OK
```

**Pull request merged**